plugin/hooks.go

```diff
@@ -270,4 +276,64 @@ type Hooks interface {
 	//
 	// Minimum server version: 7.0
 	OnCloudLimitsUpdated(limits *model.ProductLimits)
+
+	// UserHasPermissionToCollection determines if the given user has access to
+	// the given collection. Plugins are only expected to handle their own
+	// collections, and should return an error for unknown collections.
+	//
+	// For Threads Everywhere, products are expected to support at least the
+	// following permissions: create_post, edit_post, delete_post,
+	// edit_others_posts, and delete_others_posts.
+	//
+	// EXPERIMENTAL: This hook is experimental and can be changed without advance notice.
+	//
+	// Minimum server version: 7.6
+	UserHasPermissionToCollection(c *Context, user *model.User, collectionType, collectionId string, permission *model.Permission) (bool, error)
```

Comment:
After some iteration, propose just `userId` here, as it's tedious to have to pass a `*model.User` if you don't already have one, and I can't think of a reason why the plugin would need more than the `userId`.

```suggestion
UserHasPermissionToCollection(c *Context, userId, collectionType, collectionId string, permission *model.Permission) (bool, error)
```